ibm/data_source_ibm_is_bare_metal_servers.go

```diff
@@ -0,0 +1,662 @@
+										Computed:    true,
+										Description: "The unique identifier for this bare metal server disk",
+									},
+									isBareMetalServerDiskInterfaceType: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The disk interface used for attaching the disk. Supported values are [ nvme, sata ]",
+									},
+									isBareMetalServerDiskName: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The user-defined name for this disk",
+									},
+									isBareMetalServerDiskResourceType: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The resource type",
+									},
+									isBareMetalServerDiskSize: {
+										Type:        schema.TypeInt,
+										Computed:    true,
+										Description: "The size of the disk in GB (gigabytes)",
+									},
+								},
+							},
+						},
+						isBareMetalServerEnableSecureBoot: {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Indicates whether secure boot is enabled. If enabled, the image must support secure boot or the server will fail to boot.",
+						},
+						isBareMetalServerHref: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The URL for this bare metal server",
+						},
+						isBareMetalServerMemory: {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The amount of memory, truncated to whole gibibytes",
+						},
+
+						isBareMetalServerPrimaryNetworkInterface: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "Primary Network interface info",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									isBareMetalServerNicAllowIPSpoofing: {
+										Type:        schema.TypeBool,
+										Computed:    true,
+										Description: "Indicates whether IP spoofing is allowed on this interface.",
+									},
+									isBareMetalServerNicName: {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									isBareMetalServerNicPortSpeed: {
+										Type:       schema.TypeInt,
+										Computed:   true,
+										Deprecated: "This field is deprected",
+									},
+									isBareMetalServerNicHref: {
+										Type:       schema.TypeString,
+										Computed:   true,
+										Deprecated: "This URL of the interface",
+									},
+
+									isBareMetalServerNicSecurityGroups: {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+									isBareMetalServerNicSubnet: {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									isBareMetalServerNicPrimaryIP: {
```

Comment:
If the response does not have primary_ip as in scoped spec, can we remove this attribute from schema and keep only primary_ipv4_address ?